### PR TITLE
Display unsaved buffer while prompting to save.

### LINKED ISSUE #6070: Show the unsaved file while displaying the 'unsaved changes' alert at closing.
If you have several files with unsaved changes, when you try to exit the application you will only see the current open buffer when clicking discard. Atom should show you the particular buffer about to be saved or discarded.

I have atom 0.144.0 on an ubuntu 14.04

Similar problem here: #5823

### PR DESCRIPTION
**Pull request by mathiasvr**
This should fix #6070

### CODE REVIEW

**Comment by botverse** [comment]
Looks good to me
:+1:

**Comment by botverse** [comment]
Oh wait.

Is it going to show the current item after hitting `Don't save` too?
Why do you restore the original file after `Save`?

**Comment by abe33** [comment]
Having specs for this feature along with the patch would be great, so that future change don't break it.

**Comment by mathiasvr** [comment]
@botverse The file being closed will be shown while showing the dialog, if you choose to save, the file remains visible until the save dialog closes as well. After any dialog the file which was active before pressing close will be re-activated. This is of course redundant when the file being closed is already the active one, or when closing all files at once. Maybe there is a better way?

File: src/pane.coffee [hunk 1]
```diff
@@ -457,13 +457,20 @@ class Pane extends Model
     else
       return true
 
+    currentActiveItem = @activeItem
+    @activateItem item
```
Comment (mathiasvr) [inline src/pane.coffee]: This is the only line which should be merged to sublime text like functionality.

**Comment by postcasio** [comment]
If you have multiple unsaved items, will this keep reactivating the originally active item between activating each unsaved item? Ideally it should activate each item, then finally return to the original item once all unsaved items have been prompted.

Alternatively, just don't activate the original item and leave the unsaved item active. ST2 has this behaviour for what it's worth.

**Comment by botverse** [comment]
I've been playing around with the both options and I actually prefer what @mathiasvr has developed, this is for you to compare:

Only with this: `@activateItem item`

dumber

With the returning to tab function:

clever

So for me this is a :+1: as it is.

**Comment by mathiasvr** [comment]
@botverse Very good idea with the comparison :smiley: 

@postcasio Yes it will keep reactivating the original item when closing multiple items, however this is not visible to the user. It would be nice to avoid this, but then this feature would have to be spread out to multiple functions. Alternatively we could pass reactivation as an option?

**Comment by alexandernst** [comment]
Any news on this one?

**Comment by mnquintana** [comment]
@mathiasvr Can you add some specs for this?

**Comment by alexandernst** [comment]
Original dev @mathiasvr seems missing which is sad as this feature looks really awesome :(

**Comment by Zireael07** [comment]
Yeah, this would be awesome!

**Comment by mnquintana** [comment]
I might try to take this up in the next week if I have some free time :smile:

**Comment by thedaniel** [comment]
I added the `help-wanted` label in case anyone from the community wants to pick this back up.

**Comment by stereobooster** [comment]
Tried to pick this up https://github.com/atom/atom/compare/master...stereobooster:6075-focus-tab-on-save-prompt?expand=1

But for some reason Atom doesn't want to switch active tab before it shows save prompt. It treats currently destroyed buffer as active, while it is not displayed :/

**Comment by Siilwyn** [comment]
I gave it a shot but this is actually pretty hard to do now. All dialog (related) code is written synchronous and thus while the dialog is open it's not possible to switch from tab. So I tried to re-write some functions to accept callbacks but it goes a lot deeper than I initially thought...

https://github.com/atom/atom/compare/master...Siilwyn:display-unsaved-buffer

**Comment by damieng** [comment]
PR is very out of sync and hasn't been merged in 2 years.

**Pull request closed without merging**